src/Chapter3.hs

```diff
@@ -971,6 +1117,29 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek = Monday |
+                  Tuesday |
+                  Wednesday |
+                  Thursday |
+                  Friday |
+                  Saturday |
+                  Sunday
+                  deriving (Eq,Enum,Show)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend a
+          | a == Saturday || a == Sunday = True
+          | otherwise = False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay Sunday = Monday
+nextDay a = succ a
+
+daysToParty :: DayOfWeek -> Int
+daysToParty Saturday = 7
+daysToParty Sunday = 6
+daysToParty a =  (1 + (fromEnum Friday))
```

Comment:
This looks incorrect because variable `a` is not used on the right side of `=`, which means that this function returns the same number of days for all weekdays before `Friday` :thinking: